src/utilities.jl

```diff
@@ -0,0 +1,183 @@
+# This file is a part of JuliaFEM.
+# License is MIT: see https://github.com/JuliaFEM/Materials.jl/blob/master/LICENSE
+
+module Utilities
+
+using Tensors, ForwardDiff
+
+export Symm2, Symm4
+export delta, II, IT, IS, IA, IV, ID, isotropic_elasticity_tensor
+export lame, delame, debang, find_root
+
+"""Symm2{T} is an alias for SymmetricTensor{2,3,T}."""
+const Symm2{T} = SymmetricTensor{2,3,T}
+
+"""Symm4{T} is an alias for SymmetricTensor{4,3,T}."""
+const Symm4{T} = SymmetricTensor{4,3,T}
+
+"""
+    delta(i::Integer, j::Integer)
+
+Kronecker delta, defined by delta(i, j) = (i == j) ? 1 : 0.
+"""
+delta(i::T, j::T) where T <: Integer = (i == j) ? one(T) : zero(T)
+
+# TODO: We could probably remove the type argument, and just let the results be
+# inferred as Symm4{Int64}, Symm4{Rational{Int64}} and similar. Simpler to use,
+# and those behave correctly in calculations with types involving other reals
+# such as Float64.
+# Performance implications? Is the Julia compiler smart enough to optimize?
+"""
+    II(T::Type=Float64)
+
+Rank-4 unit tensor, defined by II : A = A for any rank-2 tensor A.
+"""
+II(T::Type=Float64) = Symm4{T}((i,j,k,l) -> delta(i,k)*delta(j,l))
+
+"""
+    IT(T::Type=Float64)
+
+Rank-4 unit tensor, defined by IT : A = transpose(A) for any rank-2 tensor A.
+"""
+IT(T::Type=Float64) = Symm4{T}((i,j,k,l) -> delta(i,l)*delta(j,k))
+
+"""
+    IS(T::Type=Float64)
+
+Rank-4 unit tensor, symmetric. IS ≡ (1/2) (II + IT).
+"""
+IS(T::Type=Float64) = 1//2 * (II(T) + IT(T))
+
+"""
+    IA(T::Type=Float64)
+
+Rank-4 unit tensor, screw-symmetric. IA ≡ (1/2) (II - IT).
+"""
+IA(T::Type=Float64) = 1//2 * (II(T) - IT(T))
+
+"""
+    IV(T::Type=Float64)
+
+Rank-4 unit tensor, volumetric. IS ≡ (1/3) I ⊗ I, where I is the rank-2 unit tensor.
+"""
+IV(T::Type=Float64) = Symm4{T}((i,j,k,l) -> 1//3 * delta(i,j)*delta(k,l))
+
+"""
+    ID(T::Type=Float64)
+
+Rank-4 unit tensor, deviatoric. ID ≡ IS - IV.
+"""
+ID(T::Type=Float64) = IS(T) - IV(T)
+
+"""
+    isotropic_elasticity_tensor(lambda::T, mu::T) where T <: Real
+
+Compute the elasticity tensor C(i,j,k,l) (rank 4, symmetric) for an isotropic
+material having the Lamé parameters `lambda` and `mu`.
+
+If you have (E, nu) instead, use `lame` to get (lambda, mu).
+"""
+isotropic_elasticity_tensor(lambda::T, mu::T) where T <: Real = 3 * lambda * IV(T) + 2 * mu * IS(T)
+
+"""
+    lame(E::Real, nu::Real)
```

Comment:
Perhaps add the link to the useful table found in Wikipedia:
https://en.wikipedia.org/wiki/Template:Elastic_moduli

Reply:
Yes.